tests/cache/test_util.py

```diff
@@ -0,0 +1,69 @@
+from dogpile.cache import util
+
+
+class A:
+    @classmethod
+    def class_method(cls):
+        pass
+
+    @staticmethod
+    def static_method():
+        pass
+
+    def instance_method(self):
+        pass
+
+    def nested_method(self):
+        def nested():
+            pass
+
+        return nested
+
+
+def test_function_key_generator_qualname():
```

Comment:
I would break up each test case here and have them come in using [pytest.mark.parametrize](https://docs.pytest.org/en/7.1.x/how-to/parametrize.html)  as a decorator on `def test_function_key_generator_qualname`.    so one function, but the cases come in parameters.  each parameter results in one test.